File: MyClub/MyClub.Model/Responses/DashboardRevenuePerMonthResponse.cs
using System;

namespace MyClub.Model.Responses
{
    public class DashboardRevenuePerMonthResponse
    {
        public string Month { get; set; }
        public int Year { get; set; }
        public string MonthName { get; set; }
        public decimal TotalAmount { get; set; }
        public string Currency { get; set; } = "BAM";
    }
} 

File: MyClub/UI/myclub_desktop/lib/models/dashboard/dashboard_revenue_per_month_response.dart
class DashboardRevenuePerMonthResponse {
  final String month;
  final int year;
  final String monthName;
  final double totalAmount;
  final String currency;

  DashboardRevenuePerMonthResponse({
    required this.month,
    required this.year,
    required this.monthName,
    required this.totalAmount,
    required this.currency,
  });

  factory DashboardRevenuePerMonthResponse.fromJson(Map<String, dynamic> json) {
    return DashboardRevenuePerMonthResponse(
      month: json['month'],
      year: json['year'],
      monthName: json['monthName'],
      totalAmount: json['totalAmount']?.toDouble() ?? 0.0,
      currency: json['currency'],
    );
  }
}
